nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/manager/ComponentEntityMerger.java

```diff
@@ -68,10 +75,13 @@ default void merge(final EntityType clientEntity, final Map<NodeIdentifier, Enti
             if (clientEntity.getBulletins().size() > MAX_BULLETINS_PER_COMPONENT) {
                 clientEntity.setBulletins(clientEntity.getBulletins().subList(0, MAX_BULLETINS_PER_COMPONENT));
             }
+        } else {
+            clientEntity.setBulletins(null);
```

Comment:
I think we need to continue to set the component to null when `canRead` is false. It may have been changed to accommodate an earlier iteration of this PR.

Reply:
Fixed. Thanks for caching this!